### PR TITLE
python312Packages.mygene: init at 3.2.2

### LINKED ISSUE #344328: Package request: MyGene.py
**Project description**

[MyGene.Info](http://mygene.info/) provides simple-to-use REST web services to query/retrieve gene annotation data. mygene.py is an easy-to-use Python wrapper to access [MyGene.Info](http://mygene.info/) services.

**Metadata**

* homepage URL: [https://docs.mygene.info/projects/mygene-py/en/latest/](https://docs.mygene.info/projects/mygene-py/en/latest/)
* source URL: [https://github.com/biothings/mygene.py](https://github.com/biothings/mygene.py)
* license: Redistributable (must include copyright). [Link](https://github.com/biothings/mygene.py?tab=License-1-ov-file#readme)
* platforms: All

### PR DESCRIPTION
**Pull request by rayhem**
Resolves #344328

Introduces `mygene` as an available python library to use in gene analysis.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by FliegendeWurst — APPROVED** [review]

File: pkgs/development/python-modules/mygene/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    repo = "mygene.py";
+    rev = "v${version}";
+    hash = "sha256-/KxlzOTbZTN5BA0PrJyivVFh4cLtW90/EFwczda61Tg=";
+  };
+
+  build-system = [ setuptools ];
+  dependencies = [
+    biothings-client
+    requests
+  ] ++ (if with-pandas then [ pandas ] else [ ]);
```
Comment (natsukium) [inline pkgs/development/python-modules/mygene/default.nix:27]:
```suggestion
  ];
```
Please use `optional-dependencies` if necessarry.

File: pkgs/development/python-modules/mygene/default.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  setuptools,
+  biothings-client,
+  requests,
+  with-pandas ? false,
```
Comment (natsukium) [inline pkgs/development/python-modules/mygene/default.nix:8]:
```suggestion
```

File: pkgs/development/python-modules/mygene/default.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    rev = "v${version}";
+    hash = "sha256-/KxlzOTbZTN5BA0PrJyivVFh4cLtW90/EFwczda61Tg=";
+  };
+
+  build-system = [ setuptools ];
+  dependencies = [
+    biothings-client
+    requests
+  ] ++ (if with-pandas then [ pandas ] else [ ]);
+  pythonImportsCheck = [ "mygene" ] ++ (if with-pandas then [ "pandas" ] else [ ]);
```
Comment (natsukium) [inline pkgs/development/python-modules/mygene/default.nix:28]:
```suggestion
  pythonImportsCheck = [ "mygene" ];
```

File: pkgs/development/python-modules/mygene/default.nix [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
```
Comment (natsukium) [inline pkgs/development/python-modules/mygene/default.nix:29]: Could you enable tests with `pytestCheckHook`?
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#testing-python-packages-testing-python-packages

**Review by natsukium — DISMISSED** [review]

**Comment by FliegendeWurst** [comment]
Please squash commits and drop the merge commit (rebase instead).

File: pkgs/development/python-modules/mygene/default.nix (hunk 4 above)
Comment (rayhem) [reply pkgs/development/python-modules/mygene/default.nix:35]: The tests in upstream only make requests to exercise the module (it's just a thin wrapper library to facilitate those requests). As I understand, network requests aren't available in a nix build, so there's not much testing to be done beyond ensuring the module is available.

**Comment by FliegendeWurst** [comment]
Please squash and rebase on master.

**Comment by rayhem** [comment]
Sorry for the noise @FliegendeWurst - I'm not the most familiar with git(hub; I mostly use perforce in my day-to-day). None of my logs locally showed the merge commit and pulling reported everything as up-to-date; I don't know why it persisted in this PR, but another force push to sync everything seems to have squashed correctly.

**Comment by FliegendeWurst** [comment]
I think there is a button on Github that will create a merge commit.

**Review by FliegendeWurst — APPROVED** [review]

**Pull request merged**